components/compositing/compositor.rs

```diff
@@ -1656,7 +1800,19 @@ impl<Window: WindowMethods> IOCompositor<Window> {
                 // frame tree.
                 let mut pipeline_epochs = HashMap::new();
                 for (id, details) in &self.pipeline_details {
-                    pipeline_epochs.insert(*id, details.current_epoch);
+                    if let Some(ref webrender) = self.webrender {
+                        let webrender_pipeline_id = id.to_webrender();
+                        match webrender.current_epoch(webrender_pipeline_id) {
+                            Some(epoch) => {
+                                let webrender_traits::Epoch(epoch) = epoch;
```

Comment:
Could fold this `let` into the `Some` above: `Some(webrender_traits::Epoch(epoch)) => {`